### ISSUE TITLE
Disable built-in `DocumentLinkProvider`s for custom `FileSystemProvider`s

### ISSUE THREAD

**Issue by kyswtn** [microsoft/vscode#231397]
Currently URLs and file paths are linked on all (or maybe most) documents, even on custom file system providers. I have a custom file system provider that doesn't follow UNC, so all these links are broken. It'd be great if I can turn them off using an option.

**Comment by jrieken** [comment]
> I have a custom file system provider that doesn't follow UNC, so all these links are broken.

Not sure what that means? Can you give me a sample or links that work vs don't work?

**Comment by kyswtn** [comment]
Sorry that was confusing. I meant in that file system, the files are resolved differently than you'd expect in a traditional fs. But vscode has default link provider that takes a `/abc/def` (let's say in a CSS document) and link it to `scheme://authority/abc/def` where `scheme` and `authority` are from taken the document. You see, I might not want that. 

I have no problem if they are just clickable links. But these links are delegated to `stat` method of the custom file system provider every time a document is opened, and because these links are not resolvable in the first place, users will have `stat` errors.

**Comment by jrieken** [comment]
> because these links are not resolvable in the first place, users will have stat errors.

Hm, but they should be, right? If your fs return `def` inside `abc` a stat call for `/abc/def` must resolve. It's not just links that will hit this route but many other places too, right history or any other extension that just consumes your fs via `vscode.workspace.fs.stat`

**Comment by kyswtn** [comment]
Yes, ideally, they should be. That's what I meant by not following universal naming conventions. My fs does weird translations to these referenced paths before resolution. In that scenario, being able to disable these links would be great as they're broken. What do you think?

**Comment by jrieken** [comment]
> In that scenario, being able to disable these links would be great as they're broken. What do you think?

I think your fs is broken ;-) We are [pretty clear](https://github.com/microsoft/vscode/blob/566fb7460fc6a5c44403cff02ffc9ae8db871c3f/src/vscode-dts/vscode.d.ts#L9192-L9193) about requiring hierarchical, posix'ish paths and disabling features because a fs doesn't work like this is not a fix. Also, as said before, you will encounter many different situations that can trigger this, not just links

**Comment by kyswtn** [comment]
Understood. Thanks!
